java-checks/src/main/java/org/sonar/java/checks/SunPackagesUsedCheck.java

```diff
@@ -75,6 +75,19 @@ private static boolean isSunClass(String reference) {
     return reference.startsWith("sun.");
   }
 
+  private static boolean isActuallySunPackage(MemberSelectExpressionTree tree) {
+    // Check if the expression's type actually comes from a sun.* package
+    // This prevents false positives when a variable is named "sun"
+    var type = tree.expression().symbolType();
+    if (type.isUnknown()) {
+      // If we can't determine the type, we should still check to avoid missing real issues
```

Comment:
I would argue that when in doubt we shouldn't report instead, to prevent raising FPs